lib/tasks/repositories.rake

```diff
@@ -0,0 +1,25 @@
+namespace :repo do
+  desc "Go through all non-archived repositories and open PRs with any missing contributing docs"
+  task :open_prs, [:dry_run] => :environment do |_, args|
+    dry_run = args[:dry_run].to_s.casecmp?("dry_run")
+    repos_needing_action = 0
+    opened_prs = 0
+
+    repos = Repo.non_archived
+    puts "Total #{repos.size}"
+
+    repos.each do |repo|
+      next unless repo.needs_action?
+
+      repos_needing_action += 1
+      pull_request = PullRequestCreator.new(repo.name).open_pull_request unless dry_run
```

Comment:
Again, as mentioned previously, seperating out the concerns in the `PullRequestCreator` class will make this task a lot faster, as currently we're making a whole load of unnecessary HTTP queries fetching information about a repo, after we already know about it.

Reply:
Thanks for looking at this! Do you have any time to work on it together, in case @lawrence-forooghian (who knows the arcane inner workings of graphql) is still unavailable due to client work?